Library/ENV/4.3/cc

```diff
@@ -194,6 +196,27 @@ class Cmd
   end
 
   def keep?(path)
+    # The logic in this method will eventually become the default,
+    # but is currently opt-in.
+    return keep_orig?(path) unless ENV["HOMEBREW_EXPERIMENTAL_FILTER_FLAGS_ON_DEPS"]
+
+    # first two paths: reject references to Cellar or opt paths
+    # for unspecified dependencies
+    if path.start_with?(cellar)
+      %r[#{cellar}/([\w\-_\+]+)] =~ path
```

Comment:
Pushed an update that merges these into one regex that's assigned in the class initializer rather than in `keep?`, and commented to make the capture groups clear.